src/yunohost/data_migrations/0006_migrate_pwd.py

```diff
@@ -0,0 +1,132 @@
+import spwd
+import crypt
+import random
+import string
+import subprocess
+
+from moulinette import m18n
+from moulinette.core import MoulinetteError
+from moulinette.utils.log import getActionLogger
+from moulinette.utils.process import run_commands
+from moulinette.utils.filesystem import append_to_file
+from moulinette.authenticators.ldap import Authenticator
+
+from yunohost.tools import Migration
+
+logger = getActionLogger('yunohost.migration')
+SMALL_PWD_LIST = ["yunohost", "olinux"]
+
+class MyMigration(Migration):
+    "Migrate password"
+
+    def migrate(self):
+
+        if self._is_root_pwd_listed(SMALL_PWD_LIST):
+            new_hash = self._get_admin_hash()
+            self._replace_root_hash(new_hash)
+
+    def backward(self):
+
+        pass
+
+    def _get_admin_hash(self):
+        """
+        Ask for admin hash the ldap db
+        Note: to do that like we don't know the admin password we add a second
+        password
```

Comment:
I'm testing this and : is this method really needed to get the admin hash ?

I just realized that, as root, you can run `slapcat` from which you may extract the relevant info. Here's a quick and dirty command to do that (we might want to have a more robust way to do that though) : 

```
slapcat | grep "dn: cn=admin,dc=yunohost,dc=org" -A20 | grep userPassword -A2 | tr -d '\n ' | tr ':' ' ' | awk '{print $2}' | base64 -d
```